contracts/modules/token/TokenController.sol

```diff
@@ -24,6 +25,10 @@ contract TokenController is ITokenController, LockHandler, LegacyMasterAware {
   IAssessment public assessment;
   IGovernance public governance;
 
+  ICover public cover;
+  mapping(uint => uint) stakingPoolNXMRewards;
```

Comment:
So here I actually want to verify if the caller is a staking pool.

To do this i ask the caller to provide its pool id, and then use the stakingPool() function to verifiy msg.sender is the pool id with that id.

If i don't do this, and i don't ask for the id, i'd have to keep some register of pool staking addresses.

I think this is simpler and achieves authentication. What i would think here is: is there any way to circumvent this authentication? seem like not because all valid staking pools can only be created by cover.sol